src/main/pages/che-7/hands-on-guidance/proc_adding-support-for-a-new-language.adoc

```diff
@@ -1,27 +1,38 @@
-* Delete the section title and bullets if the assembly has no prerequisites.
+* Install the latest Che-Theia plug-in Yeoman generator by running:
++
+----
+$ npm install -g yo @theia/generator-plugin
+----
 
 [discrete]
 == Procedure
 
-. Start each step with an active verb.
+. Generate a sample by running `yo @theia/plugin`. When prompted, confirm the choices for *Backend plug-in*, *Samples*, and then *Language sample*. This creates a project and a ready-to-use plug-in, which can be deployed.
 
-. Include one command or action per step.
+. Replace the download url of the implementation of language server protocol in `src/download-xml-ls.js` file. This file will be downloaded during the build process of the plug-in. If the name of `src/download-xml-ls.js` was changed, need to replace it in `package.json` in `scripts` block as well.
 
-. Use an unnumbered bullet (*) if the procedure includes only one step.
+. Replace the configuration of the language in `src/language-configuration.json` file, this file should contain configuration options for current language. If the name of `src/language-configuration.json` was changed, need to replace it in `package.json` in `contribution` block as well.
+
+. Replace the configuration of the language in `src/xml.tmLanguage.json` file, this file describes syntax highlighting, determines the color and style of source code displayed in the editor. If the name of `src/xml.tmLanguage.json` was changed, need to replace it in `package.json` in `contribution.grammars` block as well.
+
+. Replace an information about the provided language in `src/plug-in_name.ts`. In that file should be an information about the location of LSP implementation and a command with arguments to run it.
+
+. Build the plug-in by running `yarn` in the root of the project.
+
+. Deploy the Che-Theia plug-in via the plug-in registry.
 
 [discrete]
 == Additional resources
 
-* A bulleted list of links to other material closely related to the contents of the procedure module.
-* For more details on writing procedure modules, see the link:https://github.com/redhat-documentation/modular-docs#modular-documentation-reference-guide[Modular Documentation Reference Guide].
-* Use a consistent system for file names, IDs, and titles. For tips, see _Anchor Names and File Names_ in link:https://github.com/redhat-documentation/modular-docs#modular-documentation-reference-guide[Modular Documentation Reference Guide].
+* link:https://microsoft.github.io/language-server-protocol/[Language Server Protocol]
+* link:https://code.visualstudio.com/api/language-extensions/overview[VS Code Language Extensions Documentation]
+* link:https://code.visualstudio.com/api/language-extensions/language-server-extension-guide[A simple Language Server for plain text files]
+* link:adding-a-che-theia-plug-in-to-the-che-plug-in-registry.html[Adding a Che-Theia plug-in to the Che plug-in registry]
```

Comment:
This points nowhere. What's the idea?

Reply:
The correct link is `link:extending-che.html#adding-a-che-theia-plug-in-to-the-che-plug-in-registry_publishing-che-theia-plug-ins[Adding a Che-Theia plug-in to the Che plug-in registry]`

Reply:
Fixed, thanks @rkratky

Reply:
@rkratky how can we verify the correctness of that kind of link?

Reply:
@tsmaeder I've checked it on next way:
1. Go to https://www.eclipse.org/che/docs/che-7/contributor-tasks.html
2. Replace `contributor-tasks.html` with `extending-che.html#adding-a-che-theia-plug-in-to-the-che-plug-in-registry_publishing-che-theia-plug-ins`